[8.5][DOCS] Add release notes for 8.5.0 (#91029)
elastic/elasticsearch: docs/reference/release-notes/8.5.0.asciidoc

* Fix merging with empty results {es-pull}86939[#86939] (issue: {es-issue}84622[#84622])
* Fix partial reduce bug in `ip_prefix` {es-pull}89734[#89734] (issue: {es-issue}89686[#89686])

> * Remove unexpected meta param in agg's response {es-pull}89467[#89467] (issue: {es-issue}89455[#89455])

Reviewer: ```suggestion
* Remove unexpected meta parameter in aggregation response {es-pull}89467[#89467] (issue: {es-issue}89455[#89455])
```

> * Consistently allow sorting `top_metrics` by size>1 {es-pull}89974[#89974] (issue: {es-issue}86663[#86663])

Reviewer: ```suggestion
* Consistently allow sorting `top_metrics` aggregations whose size is greater than 1 {es-pull}89974[#89974] (issue: {es-issue}86663[#86663])
```

Authorization::
* Remove magic string for the `"__empty"` role {es-pull}89766[#89766]

CCR::
* Order follower backing indices for data streams by original name {es-pull}90850[#90850] (issue: {es-issue}90820[#90820])

Cluster Coordination::
* Capture deprecation warnings in batched master tasks {es-pull}85525[#85525] (issue: {es-issue}85506[#85506])
* Check circuit breaker before sending join request {es-pull}89318[#89318] (issue: {es-issue}85003[#85003])

Distributed::
* Fork `TransportClusterStateAction` to `MANAGEMENT` thread pool {es-pull}90996[#90996]

Engine::
* Fix `"path.conf'"` typo in `Security.java` {es-pull}89248[#89248] (issue: {es-issue}89327[#89327])

Geo::
* Buffer H3Polygon2D bounding boxes to avoid edge precision issues {es-pull}89196[#89196] (issues: {es-issue}89868[#89868], {es-issue}87391[#87391])
* Fix date histogram range edge case {es-pull}88957[#88957]
* Format runtime `geo_points` {es-pull}85449[#85449] (issue: {es-issue}85245[#85245])
* Generate error if scripts load `_source` with synthetic `_source` enabled {es-pull}88334[#88334]

Graph::
* Fix race condition in timeout {es-pull}88946[#88946] (issue: {es-issue}55396[#55396])

Health::

> * Disk indicator change details {es-pull}90189[#90189]

Reviewer: ```suggestion
* Modify disk indicator details to provide an overview of the disk space health {es-pull}90189[#90189]
```

* Fix disk indicator impacts and diagnosis {es-pull}90262[#90262]
* Fix the details calculation of the disk indicator {es-pull}90869[#90869]

> * Report impact and diagnosis data nodes w/o disk space and no blocked indices {es-pull}90772[#90772] (issue: {es-issue}90442[#90442])

Reviewer: ```suggestion
* Report impact and diagnosis data nodes without disk space and no blocked indices {es-pull}90772[#90772] (issue: {es-issue}90442[#90442])
```

> * Update min version for health node reporting to 8.5 {es-pull}90365[#90365] (issue: {es-issue}90359[#90359])

Reviewer: ```suggestion
* Update minimum version for health node reporting to 8.5 {es-pull}90365[#90365] (issue: {es-issue}90359[#90359])
```

Indices APIs::
* Avoid capturing per-task `RolloverResult` {es-pull}90626[#90626] (issue: {es-issue}90620[#90620])

> * Fix shard splitting for `nested` {es-pull}89351[#89351] (issue: {es-issue}88109[#88109])

Reviewer: ```suggestion
* Fix shard splitting for `nested` field type {es-pull}89351[#89351] (issue: {es-issue}88109[#88109])
```

Infra/Core::
* Fix `allow_no_indices` request option in special cases {es-pull}89622[#89622]
* Fix repeated error save loops in File Settings Service {es-pull}90271[#90271] (issue: {es-issue}90222[#90222])
* Fix date rounding for date math parsing {es-pull}90458[#90458] (issue: {es-issue}90187[#90187])
* Fix disabling APM tracing for `CancellableTask` in `TrainedModelAssignmentNodeService` {es-pull}90972[#90972] (issue: {es-issue}89850[#89850])
* Support camel case dates on 7.x indices {es-pull}88914[#88914] (issue: {es-issue}84199[#84199])

Infra/Scripting::
* Fix true/false accumulation bug in boolean `source` fallback {es-pull}90895[#90895]

Ingest Node::
* Set the enrich maintenance cluster lifecycle listener only once {es-pull}90486[#90486]

Machine Learning::
* Require correct tier processors when multiple AZs are present {es-pull}90903[#90903]

> * Return 408 instead of 500 when open/start APIs time out {es-pull}89775[#89775] (issue: {es-issue}89585[#89585])

Reviewer: ```suggestion
* Return `408` instead of `500` when open/start APIs time out {es-pull}89775[#89775] (issue: {es-issue}89585[#89585])
```

Mapping::
* Fix duplication bug for `source` fallback in numeric types {es-pull}89352[#89352]
* Include runtime fields in total fields count {es-pull}89251[#89251] (issue: {es-issue}88265[#88265])
* Fix `aggregate_metric_double` multi-values exception {es-pull}90290[#90290]

> * Validate field names when subobjects are disabled {es-pull}90950[#90950]

Reviewer: ```suggestion
* Validate field names when sub-objects are disabled {es-pull}90950[#90950]
```

Monitoring::
* Add fields to fix {ls} cgroup graphs {es-pull}90493[#90493]

Network::
* Fix `RecyclerBytesStreamOutput` allocating unlimited heap for some capacities {es-pull}90632[#90632]

Recovery::
* Fix overcounting recovered bytes after network disconnect {es-pull}90477[#90477] (issue: {es-issue}90441[#90441])

Search::

> * Add support for predefined char class regexp on wildcard fields {es-pull}90064[#90064]

Reviewer: ```suggestion
* Add support for predefined character class regular expression on wildcard fields {es-pull}90064[#90064]
```

* Safeguard `RegExp` use against `StackOverflowError` {es-pull}84624[#84624] (issue: {es-issue}82923[#82923])
* Use MB rather than GB to calculate max boolean clauses {es-pull}90309[#90309] (issue: {es-issue}86136[#86136])

Snapshot/Restore::
* Fix incorrect failed shards count in APIs for current snapshots {es-pull}89534[#89534]
* Fix over-allocation of mounted indices on a cold/frozen node {es-pull}86331[#86331]
* Fix quadratic complexity in `SnapshotStatus` serialization {es-pull}90795[#90795]
* Fork building snapshot status response off of transport thread {es-pull}90651[#90651]
* Make sure listener is resolved when file queue is cleared {es-pull}89929[#89929]
* Re-register a corrupt repository to unblock it {es-pull}89719[#89719] (issue: {es-issue}89130[#89130])
* Reject unknown request body fields in mount API {es-pull}88987[#88987] (issue: {es-issue}75982[#75982])

TSDS::
* Fix segment stats in TSDS {es-pull}89754[#89754] (issue: {es-issue}89609[#89609])
* Fix extra fields in `GET` request for synthetic `_source` {es-pull}89778[#89778]
* Fix `scaled_float` rounding for synthetic `_source` {es-pull}88916[#88916] (issue: {es-issue}88854[#88854])

Transform::

> * Don't fail a transform on a ClusterBlockException, this may be due to ILM closing an index {es-pull}90396[#90396] (issue: {es-issue}89802[#89802])

Reviewer: ```suggestion
* Retry a cluster block exception instead of failing as an error {es-pull}90396[#90396] (issue: {es-issue}89802[#89802])
```

* Fix NPE in transform scheduling {es-pull}90347[#90347] (issues: {es-issue}90356[#90356], {es-issue}88203[#88203], {es-issue}90301[#90301], {es-issue}90255[#90255])
* Improve error handling in state persistence {es-pull}88910[#88910] (issue: {es-issue}88905[#88905])
* Return `408` instead of `500` when the start API times out {es-pull}89774[#89774]

Vector Search::
* Fix bug for `kNN` with filtered aliases {es-pull}89621[#89621]

Watcher::

> * Allowing `xpack.notification.email.account.domain_allowlist` to be set dynamically {es-pull}90426[#90426] (issue: {es-issue}89913[#89913])

Reviewer: ```suggestion
* Allow `xpack.notification.email.account.domain_allowlist` to be set dynamically {es-pull}90426[#90426] (issue: {es-issue}89913[#89913])
```

> * Handling timeout exceptions on watcher startup {es-pull}90421[#90421] (issue: {es-issue}44981[#44981])

Reviewer: ```suggestion
* Handle timeout exceptions on Watcher startup {es-pull}90421[#90421] (issue: {es-issue}44981[#44981])
```

Distributed::
* Batch index delete cluster state updates {es-pull}90033[#90033] (issue: {es-issue}90022[#90022])
* Increase the minimum size of the management pool to `2` {es-pull}90193[#90193]

Health::
* Add IDs to health API diagnoses and impacts {es-pull}90072[#90072]
* Add a check to the master stability health API when there is no master and the current node is not master eligible {es-pull}89219[#89219]
* Add logic to `master_is_stable` indicator to check for discovery problems {es-pull}88020[#88020]
* Poll for cluster diagnostics information {es-pull}89014[#89014]
* Update SLM health diagnosis message to include unhealthy policy details {es-pull}89138[#89138]

Highlighting::
* Improve efficiency of `BoundedBreakIteratorScanner` fragmentation algorithm {es-pull}89041[#89041] (issues: {es-issue}73569[#73569], {es-issue}73785[#73785])

ILM+SLM::
* Add validations for the downsampling ILM action {es-pull}90295[#90295]

> * Ensure that ILM does not rollover empty indices {es-pull}89557[#89557] (issue: {es-issue}86203[#86203])

Reviewer: ```suggestion
* Ensure that ILM does not roll over empty indices {es-pull}89557[#89557] (issue: {es-issue}86203[#86203])
```

* Reuse informational message in lifecycle step {es-pull}89419[#89419]
* Move log-related logic into log block in `IndexLifecycleRunner` {es-pull}89292[#89292]

Infra/Core::

> * Add reserved snapshot/repo action {es-pull}89601[#89601]

Reviewer: ```suggestion
* Add reserved `/_snapshot/repo` file based settings {es-pull}89601[#89601]
```

* License check for user profile collaboration feature {es-pull}89990[#89990]

Machine Learning::

> * Add measure of non cache hit inference count {es-pull}90464[#90464]

Reviewer: ```suggestion
* Add measure of non-cache hit inference count {es-pull}90464[#90464]
```

Security::
* Support bulk updates of API keys {es-pull}88856[#88856]

TSDS::
* Add a TSID global ordinal to `TimeSeriesIndexSearcher` {es-pull}90035[#90035]
* Release time series data stream functionality {es-pull}90116[#90116] (issue: {es-issue}74660[#74660])
* Add synthetic `_source` support for `ignore_above` {es-pull}89466[#89466]

Vector Search::
* Support `dense_vector` {es-pull}89840[#89840]

[[regression-8.5.0]]
[float]
=== Regressions

Infra/Scripting::

> * Fix fields API Caching Regression {es-pull}90017[#90017]

Reviewer: ```suggestion
* Fix fields API caching regression {es-pull}90017[#90017]
```
